libconcurrent/concurrent/hsynch.c

```diff
@@ -22,34 +22,20 @@ RetVal HSynchApplyOp(HSynchStruct *l, HSynchThreadState *st_thread, RetVal (*sfu
     next_node->locked = true;
     next_node->completed = false;
 
-#if defined(__sun) || defined(sun)
-    schedctl_start(st_thread->schedule_control);
-#endif
     cur = (volatile HSynchNode *)SWAP(&l->Tail[node_of_thread].ptr, next_node);
     cur->arg_ret = arg;
     cur->next = (HSynchNode *)next_node;
-#if defined(__sun) || defined(sun)
-    schedctl_stop(st_thread->schedule_control);
-#endif
+
     st_thread->next_node = (HSynchNode *)cur;
 
     while (cur->locked) {                   // spinning
         if (isSystemOversubscribed())
```

Comment:
We may have to remove this if condition, since the newest implementation of the resched() function makes this check.

Reply:
So the you mean to replase
`
if (isSystemOversubscribed())
    resched();
else
    Pause();
`
with just
`resched();`
?

Reply:
Exactly

Reply:
Done